doc: add ESM code examples in url.md

aduh95 opened the pull request:
Alternative to https://github.com/nodejs/node/pull/38645, removing use of `require` or adding `import` equivalents where it makes sense.

@fisker what do you think?
<!--
Before submitting a pull request, please read
https://github.com/nodejs/node/blob/HEAD/CONTRIBUTING.md.

Commit message formatting guidelines:
https://github.com/nodejs/node/blob/HEAD/doc/guides/contributing/pull-requests.md#commit-message-guidelines

For code changes:
1. Include tests for any bug fixes or new features.
2. Update documentation if relevant.
3. Ensure that `make -j4 test` (UNIX), or `vcbuild test` (Windows) passes.

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
-->

fisker commented:
Switching to `import` is good, but I think removing `import`/`require` is bad. Like I said, it's not clear where these methods from.

When I first saw 

```js
new URL('file:///C:/path/').pathname;    // Incorrect: /C:/path/
fileURLToPath('file:///C:/path/');       // Correct:   C:\path\ (Windows)
```

I really don't know how to use `fileURLToPath`,
- Is it static method of `URL`? `URL.fileURLToPath(...)`
- Is it method of `URL` instance? `new URL(...).fileURLToPath()`
- Or the correct one `import {fileURLToPath} from 'url'`

It doesn't make sense to remove these imports. check https://nodejs.org/api/child_process.html, every example has `require('child_process')` it's very clear how to use it.

fisker commented:
`worker_threads` /`fs` has those `import`s as well, I don't think people care much it's `import` or `require`

aduh95 commented:
@fisker Fair enough, I've added the `require` calls back and added ESM examples. Would you like to add ESM examples to https://github.com/nodejs/node/pull/38645?

fisker commented:
Yes, with pleasure.

jasnell commented:
Landed in df744dbb9520
